.github/actions/setup-kubernetes/action.yaml

```diff
@@ -110,10 +110,22 @@ runs:
       kind: InitConfiguration
       nodeRegistration:
         criSocket: "unix:///var/run/crio/crio.sock"
+        kubeletExtraArgs:
+          runtime-cgroups: /system.slice/crio.service
       ---
-      kind: KubeletConfiguration
       apiVersion: kubelet.config.k8s.io/v1beta1
-      cgroupDriver: cgroupfs
+      kind: KubeletConfiguration
+      cgroupDriver: systemd
+      cgroupRoot: /
+      systemCgroups: /system.slice
+      kubeletCgroups: /system.slice/kubelet.service
+      enforceNodeAllocatable:
+      - pods
+      - system-reserved
+      systemReservedCgroup: /system.slice
+      kubeReservedCgroup: /system.slice/kubelet.service
+      systemReserved:
+        ephemeral-storage: 1Gi
```

Comment:
https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
> systemReserved is a set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G) pairs that describe resources reserved for non-kubernetes components. **Currently only cpu and memory are supported**. See http://kubernetes.io/docs/user-guide/compute-resources for more detail. Default: nil

Are you sure storage is supported? It is mentioned in kubeReserved that it is, but not in systemReserved

Reply:
I've checked and OKD uses it  as well and the code looks generic enough to propagate to node status https://github.com/kubernetes/kubernetes/blob/64af1adaceba4db8d0efdb91453bce7073973771/pkg/kubelet/nodestatus/setters.go#L344-L355

seems like a stale comment

Reply:
Looks like `system-reserved` entry is required in `enforceNodeAllocatable` to reserve it. From kubelet source:
```
if nc.EnforceNodeAllocatable.Has(kubetypes.SystemReservedEnforcementKey) {
		klog.V(2).InfoS("Enforcing system reserved on cgroup", "cgroupName", nc.SystemReservedCgroupName, "limits", nc.SystemReserved)
		if err := enforceExistingCgroup(cm.cgroupManager, cm.cgroupManager.CgroupName(nc.SystemReservedCgroupName), nc.SystemReserved); err != nil {
			message := fmt.Sprintf("Failed to enforce System Reserved Cgroup Limits on %q: %v", nc.SystemReservedCgroupName, err)
			cm.recorder.Event(nodeRef, v1.EventTypeWarning, events.FailedNodeAllocatableEnforcement, message)
			return fmt.Errorf(message)
		}
		cm.recorder.Eventf(nodeRef, v1.EventTypeNormal, events.SuccessfulNodeAllocatableEnforcement, "Updated limits on system reserved cgroup %v", nc.SystemReservedCgroupName)
	}
```

Reply:
Isn't available ephemeral storage tracked on the node, not in a cgroup?

Reply:
yep, looks like it's